### ISSUE TITLE
`EmptyArray` type

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/type-fest#929]
There is already [`EmptyObject`](https://github.com/sindresorhus/type-fest/blob/main/source/empty-object.d.ts), and I think a type for a strictly empty array would be useful.

**Comment by sindresorhus** [comment]
You could do it with `[] as const`, but it could be nice to have a type for this too, for discoverability.

We already have https://github.com/sindresorhus/type-fest/blob/main/source/non-empty-tuple.d.ts

**Comment by silverwind** [comment]
Yeah I guess this type would mostly be for completeness sake. I could also think about `EmptySet` and `EmptyMap`, but I don't want to bloat this repo too much. I guess `EmptyObject` has a reason for existence because of the mentioned problems surrounding the `{}` type that `[]` hopefully doesn't have.
